program_intent_engagement/apps/core/management/commands/bulk_insert_intents.py

```diff
@@ -0,0 +1,92 @@
+"""
+Command to create program intents
+"""
+
+import csv
+import logging
+import time
+
+from django.core.management import BaseCommand
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Command to add program intents defined in a csv file
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch_size',
+            action='store',
+            dest='batch_size',
+            type=int,
+            default=500,
+            help='Maximum number of intents to process. '
+                 'This helps avoid overloading the database while updating large amount of data.'
+        )
+        parser.add_argument(
+            '--sleep_time',
+            action='store',
+            dest='sleep_time',
+            type=int,
+            default=10,
+            help='Sleep time in seconds between update of batches'
+        )
+
+        parser.add_argument(
+            '--csv_filepath',
+            action='store',
+            dest='csv_filepath',
+            type=str,
+            help='Filepath for CSV containing program intent data'
+        )
+
+    def handle(self, *args, **options):
+        batch_size = options['batch_size']
+        sleep_time = options['sleep_time']
+        csv_filepath = options['csv_filepath']
+
+        with open(csv_filepath, 'r') as csv_file:
+            intents = []
+            lines = csv.reader(csv_file)
+            for line in lines:
+                program_uuid = line[0]
+                lms_user_id = line[1]
+                intent_reason = line[2]
+                certainty = line[3]
+                effective_time = line[4]
+
+                existing_intent = ProgramIntent.objects.filter(
+                    program_uuid=program_uuid,
+                    lms_user_id=lms_user_id,
+                    reason=intent_reason,
+                    certainty=certainty,
+                    effective_timestamp=effective_time
+                ).first()
+
+                if not existing_intent:
```

Comment:
can we use one of the or-create methods to make this simpler?

https://docs.djangoproject.com/en/dev/ref/models/querysets/#update-or-create

Reply:
Oh yeah! Looks like `get_or_create` would be a good candidate here

Reply:
Ah wait I remember why I didn't do this, I wanted to be able to bulk create attempts, which I would not be able to do via one of the `or-create` methods.

Reply:
oh those might be even more efficient 👍 

you might be able to bulk query out the possible existing ones but that might be too fat a query for django